src/app/functions/server/getRelevantGuidedSteps.test.ts

```diff
@@ -0,0 +1,834 @@
+    });
+
+    it("links to the scan if the user has not run a scan yet", () => {
+      expect(
+        getNextGuidedStep({
+          countryCode: "us",
+          latestScanData: {
+            scan: null,
+            results: [],
+          },
+          subscriberBreaches: [],
+          user: {
+            email: "arbitrary@example.com",
+          },
+        }),
+      ).toStrictEqual({
+        href: "/redesign/user/dashboard/fix/data-broker-profiles/start-free-scan",
+        id: "Scan",
+        completed: false,
+        eligible: true,
+      });
+    });
+
+    it("links to the scan if the user has a scan in progress and not all scan results are resolved", () => {
+      expect(
+        getNextGuidedStep({
+          countryCode: "us",
+          latestScanData: {
+            scan: {
+              ...completedScan.scan!,
+              onerep_scan_status: "in_progress",
+            },
+            results: [
+              createRandomScanResult({
+                status: "new",
+                manually_resolved: false,
+              }),
+            ],
+          },
+          subscriberBreaches: [],
+          user: {
+            email: "arbitrary@example.com",
+          },
+        }),
+      ).toStrictEqual({
+        href: "/redesign/user/dashboard/fix/data-broker-profiles/start-free-scan",
+        id: "Scan",
+        completed: false,
+        eligible: true,
+      });
+    });
+
+    it("links to the scan if the user has a completed scan and not all scan results are resolved", () => {
+      expect(
+        getNextGuidedStep({
+          countryCode: "us",
+          latestScanData: {
+            scan: {
+              ...completedScan.scan!,
+              onerep_scan_status: "finished",
+            },
+            results: [
+              createRandomScanResult({
+                status: "new",
+                manually_resolved: false,
+              }),
+            ],
+          },
+          subscriberBreaches: [],
+          user: {
+            email: "arbitrary@example.com",
+          },
+        }),
+      ).toStrictEqual({
+        href: "/redesign/user/dashboard/fix/data-broker-profiles/start-free-scan",
+        id: "Scan",
+        completed: false,
+        eligible: true,
+      });
+    });
+
+    it("does not link to the scan if the user has a completed scan and all scan results are being opted out", () => {
+      expect(
+        getNextGuidedStep({
+          countryCode: "us",
+          latestScanData: {
+            scan: {
+              ...completedScan.scan!,
+              onerep_scan_status: "finished",
+            },
+            results: [
+              createRandomScanResult({
+                status: "optout_in_progress",
+                manually_resolved: false,
+              }),
+            ],
+          },
+          subscriberBreaches: [],
+          user: {
+            email: "arbitrary@example.com",
+          },
+        }).id,
+      ).not.toBe("Scan");
```

Comment:
That is an excellent question, because I was in doubt which of the two I would use. The motivation I had for this was because I'm only testing whether the scan step would _not_ show up if the user had already completed it, and where it linked specifically didn't really matter. In other words, if we e.g. add another step later that all users would be redirected to, I wanted to avoid us then needing to update this otherwise-unrelated test.

But of course the flipside is that all of a sudden the structure of this test looks pretty differently from the other tests. So since this question proves that that confusion is there, I changed it in d080388f9.